File: benchmarks/stage06p_extended/corpus/interface_monomorphic/c/main.c
#include <inttypes.h>
#include <stdint.h>
#include <stdio.h>
#include <stdlib.h>
static uint64_t square(uint64_t value){return value*value;}
static uint64_t increment(uint64_t value){return value+1;}
static uint64_t run(uint64_t n){uint64_t sum=0;for(uint64_t i=0;i<n;++i)sum+=(i&1)?increment(i):square(i);return sum;}
int main(int argc,char **argv){if(argc!=2)return 2;uint64_t n=strtoull(argv[1],NULL,10);printf("%" PRIu64 "\n",run(n));return 0;}

File: benchmarks/stage06p_extended/corpus/interface_monomorphic/go/main.go
package main
import("fmt";"os";"strconv")
func square(v uint64)uint64{return v*v};func increment(v uint64)uint64{return v+1}
func run(n uint64)uint64{var sum uint64;for i:=uint64(0);i<n;i++{if i&1==0{sum+=square(i)}else{sum+=increment(i)}};return sum}
func main(){n,e:=strconv.ParseUint(os.Args[1],10,64);if e!=nil{panic(e)};fmt.Printf("%d\n",run(n))}
